sbws/util/stem.py

```diff
@@ -178,25 +181,33 @@ def launch_tor(conf):
                 existing_val.append(value)
                 torrc.update({key: existing_val})
     # Finally launch Tor
-    stem.process.launch_tor_with_config(
-        torrc, init_msg_handler=log.debug, take_ownership=True)
+    try:
+        stem.process.launch_tor_with_config(
+            torrc, init_msg_handler=log.debug, take_ownership=True)
+    except Exception as e:
+        fail_hard('Error trying to launch tor: %s', e)
     # And return a controller to it
     cont = _init_controller_socket(conf['tor']['control_socket'])
-    assert is_controller_okay(cont)
     # Because we build things by hand and can't set these before Tor bootstraps
     cont.set_conf('__DisablePredictedCircuits', '1')
     cont.set_conf('__LeaveStreamsUnattached', '1')
-    log.info('Started and connected to Tor %s via %s', cont.get_version(),
-             conf['tor']['control_socket'])
-    return cont
+    try:
+        log.info('Started and connected to Tor %s via %s', cont.get_version(),
+                 conf['tor']['control_socket'])
+        return cont
+    except Exception as e:
+        log.exception("Exception trying to launch tor %s", e)
```

Comment:
- If we get here, we didn't have an exception while trying to launch Tor. We got it while trying to query the control port for its version. So the error message is misleading.
- I don't think we need this try/except. We don't use a try/except every single time we use the controller (see how we don't wrap the `set_conf` calls just above this with try/except).

So I'd remove this try/except.

Reply:
You're right that is not the correct place to catch it, and it should be in ``set_conf``, but i still think we should keep it, because it's the exception in which i run running 2 scanners with the same dir, and that mistake could happen to someone else.
I probably should add to the log something like: "Maybe other instance is using the same the same tor directory?"